src/main/java/array/MainObject_Advance.java

```diff
@@ -173,35 +173,26 @@ public static void main(String[] args) {
         }
     }
 
-
+    //<editor-fold desc="Private functions">
     private static void showInfoBasic(Account account) {
         printSeparatedLine(50, "=");
-        System.out.println("Ho ten: " + account.getFirstName() + " " + account.getLastName());
-        System.out.println("Ngay tao: " + formatDate(account.getCreatedDate()));
+        System.out.println("Họ tên: " + account.getFirstName() + " " + account.getLastName());
+        System.out.println("Ngày tạo: " + formatDate(account.getCreatedDate()));
         System.out.println("Số tiền: " + formatCurrency(account.getCurrentBalance()));
         System.out.println("Loại kỳ hạn: " + account.getPeriod());
     }
 
-    private static void showInfo(Account account){
-     //   printSeparatedLine(50, "=");
-        System.out.println("Ho ten: " + account.getFirstName() + " " + account.getLastName());
-        System.out.println("Ngay tao: " + formatDate(account.getCreatedDate()));
-        System.out.println("Số tiền: " + formatCurrency(account.getCurrentBalance()));
-        System.out.println("Loại kỳ hạn: " + account.getPeriod());
-        System.out.println("Lịch sử giao dịch: ");
-        if (account.getHistories().size() == 0) {
-            System.out.println("Tài khoản chưa thực hiện bất kì giao dịch nào");
+    private static void showInfo(Account account, AccountHistory.Type type) {
+        List<AccountHistory> listAccountHistory = account.getHistories().stream().filter(transferIn ->
+                transferIn.getType() == type).collect(Collectors.toList());
+        showInfo(account, listAccountHistory);
+    }
 
-        }
-        for (AccountHistory history : account.getHistories()) {
-            dash();
-            System.out.println("Ngày tạo: " + formatDate(history.getCreatedDate(), "dd-MM-yyyy hh:mm:ss"));
-            System.out.println("Số tiền: " + formatCurrency(history.getBalance()));
-            System.out.println("Loại giao dịch: " + history.translatedType());
-        }
-        printSeparatedLine(50, "=");
+    private static void showInfo(Account account){
```

Comment:
Nếu muốn in ra thông tin tài khoản và tất cả lịch sử giao dịch của tài khoản đó thì đơn giản là gọi hàm showInfo và truyền vào 2 tham số:
- account
- listHistory của account đó, do mình ko cần chỉ định in ra danh sách lịch sử giao dịch nào nên cứ truyền vào hết thì hàm đc gọi sẽ in ra hết thôi.